### PR TITLE
Update USERS.md to add GitLab

### PR DESCRIPTION
**Pull request by whaber**
Added GitLab to the list

cc @ap4y

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

```release-note
Added GitLab to the users list.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10485)

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -65,6 +65,12 @@ Users (Alphabetically)
       D: Datadog is using Cilium in AWS (self-hosted k8s)
       U: ENI Networking, Service load-balancing, Encryption
       Q: @lbernail, @roboll
+      
+    * N: GitLab
+      D: GitLab is using Cilium as a method for implementing network policies inside Auto DevOps deployed clusters for GitLab customers using k8s
```
Comment (pchaigno) [inline USERS.md]: Maybe we can shorten this a bit into `GitLab is using Cilium to implement network policies inside Auto DevOps deployed clusters for GitLab customers using k8s`?

Comment (whaber) [reply USERS.md]: 👍 I will make that change.

Comment (whaber) [reply USERS.md]: Updated: https://github.com/cilium/cilium/pull/10485/commits/c03ec7ae0783a4153b814e2ade9de9ea8e79b851

**Comment by aanm** [comment]
@whaber can you squash both commits and add a sign-off message? https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#developer-s-certificate-of-origin

**Pull request closed without merging**